source/dub/compilers/dmd.d

```diff
@@ -172,8 +172,10 @@ class DmdCompiler : Compiler {
 				settings.addDFlags("-lib");
 				break;
 			case TargetType.dynamicLibrary:
-				version (Windows) settings.addDFlags("-shared");
-				else settings.addDFlags("-shared", "-defaultlib=libphobos2.so");
+				version(linux)
+					settings.addDFlags("-shared", "-defaultlib=libphobos2.so");
+				else
+					settings.addDFlags("-shared");
```

Comment:
You're conflating 2 different things here, building an isolated shared D library that is statically linked against phobos, and building a shared D library that is linked against a shared phobos.
I think we need something along the line of https://github.com/MartinNowak/dub/commit/bfcc6ab3b2cb3bacb1bd090d4cdc5459b865b71b instead.